src/mlpack/methods/ann/loss_functions/cross_entropy_error_impl.hpp

```diff
@@ -32,8 +35,19 @@ CrossEntropyError<InputDataType, OutputDataType>::Forward(
     const InputType& input,
     const TargetType& target)
 {
-  return -arma::accu(target % arma::log(input + eps) +
-      (1. - target) % arma::log(1. - input + eps));
+  typedef typename InputType::elem_type ElemType;
+  InputType error = target % arma::log(input + eps) +
+      (1. - target) % arma::log(1. - input + eps);
+  // Not really happy with this implementation. I am going to change this.
+  // suggestions are welcome (Without using for loops). Will figure it out tomorrow then.
+  if (numClasses)
+  {
+    for (size_t i = 0; i < input.n_elem; i++)
+    {
+      error(i) *= weights(target(i));
+    }
+  }
+  return arma::accu(error);
 }
```

Comment:
Is the minus sign missing here or this is intentional change?

Reply:
Yep missed it by mistake. I haven't tested it yet that's why I opened a draft PR.